native/cocos/scene/Model.cpp

```diff
@@ -167,6 +167,10 @@ void Model::updateUBOs(uint32_t stamp) {
 
     updateSHUBOs();
 
+    const auto *pipeline = Root::getInstance()->getPipeline();
+    const auto *shadowInfo = pipeline->getPipelineSceneData()->getShadows();
+    const auto forceUpdateUBO = shadowInfo->isEnabled() && shadowInfo->getType() == ShadowType::PLANAR;
```

Comment:
Can `shadowInfo ` be nullptr?

Reply:
pipelineSceneData will exist if it is not destroyed.